Enable Support foundation ID in service-broker routes to identify cloud foundry instance

spiegela opened the pull request:
This PR adds new routes to instance & binding controllers to identify a Cloud Foundry instance using a foundation-administrator provided ID.  The foundation ID is enabled by registering the service broker using the following syntax:

```shell-session
$ cf create-service-broker <broker-name> <username> <password> https://<broker-url>/<foundation-id>
```

This PR maintains backward compatibility. If a broker is registered "normally", the foundationId attribute will simply be set to null in the request.

This PR also closes #15 

Thanks!

scottfrederick commented on src/main/java/org/springframework/cloud/servicebroker/model/CreateServiceInstanceBindingRequest.java:
```diff
@@ -74,6 +74,12 @@
 	private transient String serviceInstanceId;
 
 	/**
+	 * The Cloud Controller GUID of the service instance to being bound.
```
this comment is incorrect

scottfrederick commented:
@spiegela This looks good, with a few comments in the commits. If you can address the comments and then squash the commits down to a small number of atomic commits, it will be ready to merge. 

Do you have a [Spring CLA](https://github.com/spring-cloud/spring-cloud-cloudfoundry-service-broker#contribute) on file?

spiegela commented:
@scottfrederick Thanks! I only saw one comment on the commits, which I've corrected.  Let me know if I've missed something.

I've squashed the commits and rebased on master.  I'm not sure what the conflicts are that Github is finding.

This is my first Spring commit, so I've just submitted CLA.  Let me know if there's any info you need from me, like my confirmation number.

scottfrederick commented:
@spiegela I'm preparing a `1.0.1` release that includes this support for registering a broker to multiple CF deployments. I'd like the change the name of the fields in the model objects from `foundationId` to `cfInstanceId`, as "foundation" is a bit of a Pivotal-specific term and I'd rather use a term that matches what's used in the [CF open-source docs](http://docs.cloudfoundry.org/services/supporting-multiple-cf-instances.html). Do you have any concerns about that?

spiegela commented:
@scottfrederick No, I have no issues with that. I'll update my broker to match when I upgrade to `1.0.1`.  Thanks for the heads up!
